spec/safe_statements_spec.rb

```diff
@@ -1,18 +1,26 @@
 require "spec_helper"
 
 RSpec.describe PgHaMigrations::SafeStatements do
-  TableLock = Struct.new(:table, :lock_type, :granted)
+  TableLock = Struct.new(:table, :lock_type, :granted, :pid)
   def locks_for_table(table, connection:)
+    identifiers = table.to_s.split(".")
+
+    identifiers.prepend("public") if identifiers.size == 1
+
+    schema, table = identifiers
+
     values = connection.execute(<<-SQL)
-      SELECT pg_class.relname AS table, pg_locks.mode AS lock_type, granted
+      SELECT pg_class.relname AS table, pg_locks.mode AS lock_type, granted, pid
       FROM pg_locks
       JOIN pg_class ON pg_locks.relation = pg_class.oid
+      JOIN pg_namespace ON pg_class.relnamespace = pg_namespace.oid
       WHERE pid IS DISTINCT FROM pg_backend_pid()
-        AND pg_class.relkind = 'r'
+        AND pg_class.relkind IN ('r', 'p')
```

Comment:
```suggestion
        AND pg_class.relkind IN ('r', 'p') -- 'r' is a standard table; 'p' is a partition parent
```

Reply:
Addressed in https://github.com/braintree/pg_ha_migrations/pull/87/commits/3d164e2cf3bced19fd3aca90fa360ede80d7c9db